backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java

```diff
@@ -67,35 +73,49 @@ public void sendVerificationEmail(String emailAddress) {
       throw new UserException(UserExceptionType.USER_ALREADY_REGISTERED);
     }
 
-    String token = emailUtils.generateJwtEmailToken(emailAddress);
+    String token = JwtUtils.generateEmailToken(emailAddress, JwtUtils.EMAIL_VERIFICATION_PURPOSE);
+
+    Map<String, String> arguments =
+        Map.of("emailAddress", emailAddress, "link", UrlUtil.getEmailVerificationLink(token));
 
-    String confirmationLink = emailUtils.getEmailConfirmationLink() + token;
-    String emailBody = emailUtils.buildVerificationEmail(emailAddress, confirmationLink);
-    Email emailToSend = new Email(emailAddress, SUBJECT_FOR_EMAIL_CONFIRMATION, emailBody, true);
-    emailService.send(emailToSend);
+    emailService.send(
+        new Email(
+            emailAddress,
+            EmailUtils.EMAIL_VERIFICATION_SUBJECT,
+            EmailUtils.EMAIL_VERIFICATION_TEMPLATE,
+            arguments,
+            true));
   }
 
   @Override
-  public void sendPasswordResetEmail(String email) {
-    if (!userRepository.existsByEmail(email)) {
+  public void sendPasswordResetEmail(String emailAddress) {
+    if (!userRepository.existsByEmail(emailAddress)) {
       throw new UserException(UserExceptionType.USER_NOT_FOUND);
     }
 
-    String token = emailUtils.generateJwtEmailToken(email);
+    String token = JwtUtils.generateEmailToken(emailAddress, JwtUtils.PASSWORD_RESET_PURPOSE);
 
-    String confirmationLink = emailUtils.getPasswordResetLink() + token;
-    String emailBody = emailUtils.buildPasswordResetEmail(email, confirmationLink);
-    Email emailToSend = new Email(email, SUBJECT_FOR_PASSWORD_RESET, emailBody, true);
-    emailService.send(emailToSend);
+    Map<String, String> arguments =
+        Map.of("emailAddress", emailAddress, "link", UrlUtil.getPasswordResetLink(token));
+
+    emailService.send(
+        new Email(
+            emailAddress,
+            EmailUtils.PASSWORD_RESET_SUBJECT,
+            EmailUtils.PASSWORD_RESET_TEMPLATE,
+            arguments,
+            true));
   }
 
   @Override
-  public String resetPassword(String token, String password) {
-    emailUtils.validateJwtToken(token);
-    String email = emailUtils.getEmailFromJwtEmailToken(token);
+  public String resetPassword(UserPasswordResetRequest userPasswordResetRequest) {
+    JwtUtils.validateEmailToken(
```

Comment:
Maybe you can provide static method that will call this method under the hood, for example JwtUtils.validateTokenForPasswordReset

Reply:
Agreed, so the logic doesn't leak into UserService.